fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt

```diff
@@ -97,42 +128,57 @@ fun TabsTray(
             ) { position ->
                 when (Page.positionToPage(position)) {
                     Page.NormalTabs -> {
-                        if (displayTabsInGrid) {
-                            TabGrid(
-                                tabs = normalTabs,
-                                onTabClose = onTabClose,
-                                onTabMediaClick = onTabMediaClick,
-                                onTabClick = handleTabClick,
-                                onTabLongClick = onTabLongClick,
-                            )
+                        val showInactiveTabsDialog = showInactiveTabsAutoCloseDialog(inactiveTabs.size)
+                        var showAutoCloseDialog by remember { mutableStateOf(showInactiveTabsDialog) }
```

Comment:
This function is starting to feel a little crowded, and I'm wondering if it would make sense/be possible to move this dialog logic into `InactiveTabsList`. It might allow us to shrink some of these long parameter lists too, but I don't know if this state needs to memoized at this level for this to work

Reply:
I'll look into reorganizing it, now that I've figured out a good pattern to use stores + Compose.

Reply:
so the main trade off I can foresee is that by ditching the parameters from `InactiveTabList` and moving the logic out of `InactiveTabsViewHolder` and `TabsTray`, is that two stores need to be passed-down. I think this will be inevitable if we ditch interactors, but it also requires the previews to instantiate these large objects.

Not a deal breaker by any means, but just some quick thoughts. I'll have to spin off a different ticket to handle this refactor though.

Reply:
Would it make sense to pass the entire Composable in? something like:

```
fun TabsTray(
    ...
    inactiveTabsContent: @Composable () -> Unit
)
```

There's lots of examples in Compose of this kind of "slot API" like in `Button`, `Scaffold`, etc. I think the question is whether this would be an appropriate use case for that kind of thing

Reply:
We could potentially do this for the banner as well, and help alleviate the concerns you shared above about the parameter list getting even larger.

Reply:
I think it'd be strange to include a Composable as a lambda parameter in this situation, when it'd only be serving the purpose of abstracting out some interactor lambdas, since `TabsTray` won't be used anywhere else in Fenix.

I think it's overall a larger architectural problem to solve of interactors + stores + data flow.

What do we pass for responding to user actions?

Do we only have one store to pass in? Do we sacrifice one degree of testing, if the actions are sent directly from the Composable?

How much data do we pass down to each node of the Composable? Just the raw data it needs? An entire store?

Reply:
Actually, I'd be greatly interested in revisiting this if the [interactor RFC](https://github.com/mozilla-mobile/firefox-android/pull/1466) becomes a thing. It would greatly simplify the lambdas that have to be passed-in, as well as reduce the business logic duplication inside of the previews.